beangulp/testing.py

```diff
@@ -76,90 +122,72 @@ def log(msg, level=0, **kwargs):
 @click.option('--verbose', '-v', count=True, help="Enable verbose output.")
 @click.option('--quiet', '-q', count=True, help="Suppress all output.")
 @click.option('--exitfirst', '-x', is_flag=True, help="Stop at the first test failure.")
 @click.pass_obj
-def _test(ctx, documents, expected, generate, verbose, quiet, exitfirst):
+def _test(ctx, documents: List[str], generate: bool,
+          verbose: bool, quiet: bool, exitfirst: bool):
     """Run the importer on all DOCUMENTS and verify that it produces the
     desired output.  The desired output is stored in Beancount ledger
     files with an header containing additional metadata.  These files
     are named as the SHA1 digest of the documents and can be generated
     with the --generate option.
 
-    DOCUMENTS can be files or directories.  Directories are walked and
-    the importer is called on all files.  When no DOCUMENTS path is
-    specified on the command line, the tests/documents/ directory
-    relative to the test script location is used.  All documents for
-    which an expected output file exist must be positively identified
-    by the importer.
-
-    Expected output files are searched in the tests/expected/$NAME/
-    directory relative to the test script location, where $NAME is
-    replaced with the importer name, if the path exists, or in the
-    tests/expected/ directory.  This path can be overwritten through
-    the --expected DIR option.
-
+    DOCUMENTS can be files or directories. Directories are walked and the
+    importer is called on all files found, except for .beancount files.
     """
     assert len(ctx.importers) == 1
     importer = ctx.importers[0]
 
-    import __main__
-    root = os.path.dirname(__main__.__file__)
-
-    if not documents:
-        documents = (os.path.join(root, 'tests','documents'), )
-
-    if not expected:
-        expected = os.path.join(root, 'tests', 'expected', importer.name())
-        if not os.path.exists(expected):
-            expected = os.path.join(root, 'tests', 'expected')
-
     verbosity = verbose - quiet
     log = logger(verbosity)
     failures = 0
 
-    for doc in walk(documents):
-        # unless verbose mode is enabled, do not output a newline so
+    for doc in walk(documents, r".*\.beancount$"):
+        # Unless verbose mode is enabled, do not output a newline so
         # the test result is printed on the same line as the test
         # document filename.
-        log(f'∙ {os.path.relpath(doc):}', nl=verbosity > 0)
-
-        # compute the path to the expected output file
-        path = os.path.join(expected, sha1sum(doc) + '.beancount')
-        f = cache.get_file(os.path.abspath(doc))
-
-        if importer.identify(f):
-            account, date, name, entries = run_importer(importer, f)
-            log(f'  {os.path.relpath(path):}', 1)
-            log(f'  {account.replace(":", "/"):}/{date:%Y-%m-%d}-{name:}', 1)
+        log(f'* {doc}', nl=verbosity > 0)
```

Comment:
Referring to your email about removing unicode to support non-unicode systems: do we really need to care about those? I think Python 3 is "unicode by default". The only issues could be printing unicode on the Windows console, and Click solves that for us, and font coverage on old terminals. I think that this is not a problem for terminal emulators in the last 20 years. The only think I am not sure about is unicode coverage of the font used by the Linux console. Summarizing: I think using characters outside ascii allows for nicer interfaces and there is no real technical reason to stick to ascii only. However, I can live with this change.

Reply:
I'm with you, but inasmuch as it's free in cost - like here, a cosmetic nicety - I see no reason to make the program depend on it.